### PR TITLE
enable package signing

### PR DESCRIPTION
**Pull request by Avinash95**
Signed-off-by: Avinash Tahakik avinashtahakik@gmail.com

### CODE REVIEW

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

File: Makefile [hunk 1]
```diff
@@ -11,6 +11,8 @@ clean: clean_openwrt clean_feeds clean_binaries
 
 # Building OpenWRT
 $(DIR__CI)/patched:
+	/vault read -field=key secret/creator/packagesigning > $(DIR__OPENWRT)/key.pem
```
Comment (abhijit-mahajani-imgtec) [inline Makefile]: should we also remove the key.pem,cert.pem during clean rule? as we are removing the .patched file too?
If key.pem/cert.pem has speciific time expiry then downloading new key/cert is useful

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

File: Makefile [hunk 2]
```diff
@@ -7,10 +7,12 @@ all: openwrt
 openwrt: build_openwrt
 	echo "OpenWrt Done!"
 
-clean: clean_openwrt clean_feeds clean_binaries
+clean: clean_openwrt clean_feeds clean_binaries clean_keys
 
 # Building OpenWRT
 $(DIR__CI)/patched:
+	/vault read -field=key secret/creator/packagesigning > $(DIR__OPENWRT)/key.pem
+	/vault read -field=cert secret/creator/packagesigning > $(DIR__OPENWRT)/cert.pem
```
Comment (nikhil-zinjurde-imgtec) [inline Makefile]: Please move the vault instructions after ./scripts/feeds install -a step. The build is failing with the following error due to current sequence

```
/vault read -field=cert secret/creator/packagesigning > <http://10.100.124.153:8080/job/Build_All_CA_PR/ws/openwrt/cert.pem>
git submodule init openwrt;git submodule update --remote; \
    cd <http://10.100.124.153:8080/job/Build_All_CA_PR/ws/openwrt;> \
    ./scripts/feeds update -a; \
    ./scripts/feeds install -a;
Submodule 'openwrt' (https://github.com/CreatorDev/openwrt.git) registered for path 'openwrt'
fatal: destination path 'openwrt' already exists and is not an empty directory.
Clone of 'https://github.com/CreatorDev/openwrt.git' into submodule path 'openwrt' failed
/bin/sh: 3: ./scripts/feeds: not found
/bin/sh: 4: ./scripts/feeds: not found
```

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

File: Makefile [hunk 3]
```diff
@@ -7,14 +7,16 @@ all: openwrt
...
 
-clean: clean_openwrt clean_feeds clean_binaries
+clean: clean_openwrt clean_feeds clean_binaries clean_keys
 
 # Building OpenWRT
 $(DIR__CI)/patched:
 	git submodule init openwrt;git submodule update --remote; \
 	cd $(DIR__OPENWRT); \
 	./scripts/feeds update -a; \
 	./scripts/feeds install -a;
```
Comment (nikhil-zinjurde-imgtec) [inline Makefile]: Think we need to add \ after each command?

```
./scripts/feeds update -a; \
./scripts/feeds install -a; \
/vault read -field=key secret/creator/packagesigning > key.pem; \
/vault read -field=cert secret/creator/packagesigning > cert.pem;
```

File: Makefile [hunk 4]
```diff
@@ -67,3 +69,8 @@ clean_feeds: clean_patches
 clean_binaries:
 	rm -rf $(DIR__OPENWRT)/bin/pistachio/
 
+.PHONY: clean_keys
+clean_keys:
+	rm -f $(DIR__OPENWRT)/key.pem
+	rm -f $(DIR__OPENWRT)/cert.pem
```
Comment (nikhil-zinjurde-imgtec) [inline Makefile]: rm -f $(DIR_OPENWRT)/key.pem; \
rm -f $(DIR_OPENWRT)/cert.pem;

File: creator-platform-all-cascoda.config [hunk 5]
```diff
@@ -10,6 +10,10 @@ CONFIG_KERNEL_KALLSYMS=y
 CONFIG_IMAGEOPT=y
 CONFIG_LOCALMIRROR="https://downloads.creatordev.io/pistachio/marduk/dl"
 CONFIG_OPENSSL_WITH_EC=y
+CONFIG_OPKGSMIME_CERT="$(TOPDIR)/cert.pem"
```
Comment (abhijit-mahajani-imgtec) [inline creator-platform-all-cascoda.config:13]: CONFIG_SMIMEOPT=y is needed to enable this package signing by key/certificates.

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

**Comment by nikhil-zinjurde-imgtec** [comment]
build all

**Comment by nikhil-zinjurde-imgtec** [comment]
Approved & verified

**Pull request merged**